geopandas/tools/sjoin.py

```diff
@@ -28,7 +31,18 @@ def sjoin(
         Suffix to apply to overlapping column names (left GeoDataFrame).
     rsuffix : string, default 'right'
         Suffix to apply to overlapping column names (right GeoDataFrame).
+    """
+    _basic_checks(left_df, right_df, how, op, lsuffix, rsuffix)
 
+    indices = _geom_predicate_query(left_df, right_df, op)
+
+    joined = _frame_join(indices, left_df, right_df, how, lsuffix, rsuffix)
+
+    return joined
+
+
+def _basic_checks(left_df, right_df, how, op, lsuffix, rsuffix):
+    """Checks validity of input parameters.
```

Comment:
Please consider expanding the docstring (feel free to edit):

```
Checks the validity of join input parameters.  
`how` and `op` must be one of the valid options.  
`'index_'` concatenated with `lsuffix` or `rsuffix` must not already exist as columns in the left or right data frames.

Parameters
------------
left_df : GeoDataFrame
right_df : GeoData Frame
how : str, one of 'left', 'right', 'inner'
    join type
op : str, one of 'intersects', 'contains', 'within'
    predicate operation used in join
lsuffix : str
   left index suffix
rsuffix : str
   right index suffix
```

Reply:
Will do.

Reply:
Expanded docstring, resolving.